terraform/gitlab/gitlab.tf.json.template.py

```diff
@@ -1497,6 +1500,42 @@ def qq(*words):
                 'depends_on': ['aws_iam_service_linked_role.gitlab_ssm']
             }
         },
+        'aws_cloudformation_stack': {
+            # Images whose short name begins with '_' are only used outside the
+            # security boundary, so vulnerabilities that are detected within
+            # them do not need to be addressed with the same urgency.
+            #
+            # Using CF stack because the AWS provider does not support
+            # Inspector filters and the AWSCC provider fails due to
+            # https://github.com/hashicorp/terraform-provider-awscc/issues/1364
+            'inspector_filters': {
+                'name': config.qualified_resource_name('inspectorfilters'),
+                'template_body': json.dumps({
+                    'AWSTemplateFormatVersion': '2010-09-09',
+                    'Description': 'Create suppression rules for select containers in AWS inspector',
+                    'Resources': {
+                        # Must be alphanumeric
+                        short_name.replace('_', ''): {
```

Comment:
This mapping is surjective which is risky. Take a look at how hashing is used in the shared component to name image-related resources.

Reply:
Hashing does not work in this context because the hash may still contain the characters `-` and `_`, even if we remove the prefix.

Attempt:
diff```
$ git diff
diff --git a/terraform/shared/shared.tf.json.template.py b/terraform/shared/shared.tf.json.template.py
index 6c81e243..4ad857a3 100644
--- a/terraform/shared/shared.tf.json.template.py
+++ b/terraform/shared/shared.tf.json.template.py
@@ -396,7 +396,7 @@ tf_config = {
                     'Description': 'Create suppression rules for select Docker images in AWS Inspector',
                     'Resources': {
                         # Must be alphanumeric
-                        short_name.replace('_', ''): {
+                        ImageRef.parse(full_name).tf_repository.removeprefix('repository_'): {
                             'Type': 'AWS::InspectorV2::Filter',
                             'Properties': {
                                 'Name': 'exclude_image' + short_name,
```

Result:

```
aws_cloudformation_stack.inspector_filters: Creating...
╷
│ Error: creating CloudFormation Stack (azul-inspectorfilters-anvildev): ValidationError: Template format error: Resource name -c_vp5z2sHBUIpD0KsPiris41Q4 is non alphanumeric.
│       status code: 400, request id: 540300ea-6dd0-43ca-9a24-a56a7789a480
│ 
│   with aws_cloudformation_stack.inspector_filters,
│   on shared.tf.json line 292, in resource[9].aws_cloudformation_stack[0].inspector_filters:
│  292:                     }
│ 
╵
make: *** [Makefile:37: apply] Error 1
```

Reply:
Ah, it does work with a different hashing representation though.